cms/models.py

```diff
@@ -325,13 +325,20 @@ def program_certificate(
 
         # Get a CertificatePage to serve this request
         certificate_page = (
-            certificate.program.page.certificate_page
-            if certificate.program.page
-            else None
+            certificate.certificate_page_revision.as_page_object()
+            if certificate.certificate_page_revision
+            else (
+                certificate.program.page.certificate_page
+                if certificate.program.page
+                else None
+            )
         )
         if not certificate_page:
             raise Http404()
 
+        if not certificate.certificate_page_revision:
+            certificate.save()
```

Comment:
Can you add a comment here mentioning what the `.save()` would be doing behind the scenes e.g. it will be associating the certificate revision for that certificate etc?

Reply:
fixed